clients/da-vinci-client/src/main/java/com/linkedin/davinci/consumer/VeniceChangeCoordinate.java

```diff
@@ -231,7 +237,7 @@ public int comparePosition(VeniceChangeCoordinate other) {
       // but it DOESNT account for version rollbacks. This should be removed once consumer sequence id is everywhere.
       return topic.compareTo(other.topic);
     }
-    return PubSubUtil.comparePubSubPositions(pubSubPosition, other.pubSubPosition);
+    return pubSubConsumer.positionDifference(topicPartition, pubSubPosition, other.pubSubPosition);
```

Comment:
Not need to happen in this PR though. I am wondering if we want to consolidate ^ sanity check logic into the implementation of the position diff?

Reply:
But consumerSequenceId/topic are only part of VCC not part of PubSubPosition

Reply:
My thinking originally was: topic and partition sanity check is important for position comparison because it makes no sense to compare position with different topic/partition.
Sequence id is VCC concept yes.
We can claim that it is user's responsibility to perform sanity check before invoking this method though, just as before when using numeric offset, but I was just wondering if we wrap this sanity check can be more helpful.
Still, can leave this out of the scope of this PR